### PR TITLE
Add massive reorg cucumber test

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
- Added two massive reorg tests greater than the  `orphan_storage_capacity`
  - _Massive reorg simple case_
  - _Massive multiple reorg_
- Fixed a bug where all orphans are cleared out every time  block sync occurs, instead of only trimming it to the specified `orphan_storage_capacity`
- Fixed some failing cucumber test scripts

**Note:** Both tests fail for blockchain length of 500+ blocks if only one node is used to connect to the fork chain; other than what is described in the log file, the cause has not been found yet:

``` rust
2021-03-23 06:23:22.354303100 [c::cs::lmdb_db::lmdb_db] ERROR Failed to apply DB transaction: You tried to execute an invalid Database operation: There is a different header stored at height 1 already. New header (99b94cfaf67093c69bb514a6f99f80dba6d175ad97a2613c3aad6b7aae1159b0), current header: (d88e1c1be6581314a32e7bcc5b901699f8de757b395d1daaefefd12371a18a29)
2021-03-23 06:23:22.354467500 [c::bn::header_sync] DEBUG Failed to synchronize headers from peer `2333e6514823aa73efae2c81ae`: Chain storage error: You tried to execute an invalid Database operation: There is a different header stored at height 1 already. New header (99b94cfaf67093c69bb514a6f99f80dba6d175ad97a2613c3aad6b7aae1159b0), current header: (d88e1c1be6581314a32e7bcc5b901699f8de757b395d1daaefefd12371a18a29)
2021-03-23 06:23:22.354497500 [c::bn::header_sync] WARN  Banned sync peer because Failed to synchronize headers from peer: Chain storage error: You tried to execute an invalid Database operation: There is a different header stored at height 1 already. New header (99b94cfaf67093c69bb514a6f99f80dba6d175ad97a2613c3aad6b7aae1159b0), current header: (d88e1c1be6581314a32e7bcc5b901699f8de757b395d1daaefefd12371a18a29)
```

## Motivation and Context
This test was created to #2253

## How Has This Been Tested?
Cucumber test

## Types of changes

* [X] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [X] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch.
* [X] I ran `cargo-fmt --all` before pushing.
* [X] I ran `cargo test` successfully before submitting my PR.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: integration_tests/log4rs_integration_tests.yml [hunk 1]
```diff
@@ -102,68 +102,109 @@ appenders:
...
+    appenders:
+      - core
+    additive: false
+  tari::base_node::grpc:
+    level: trace
+    appenders:
+      - core
     additive: false
-  # Route log events sent to the "wallet" logger to the "base_layer" appender
+  # base_layer (mainly wallet)
```
Comment (philipr-za) [inline integration_tests/log4rs_integration_tests.yml]: Feels weird to have the wallet be Base Layer but not core? If you are going to the trouble to relabel these various appenders then perhaps we should rename base_layer to wallet?

Comment (hansieodendaal) [reply integration_tests/log4rs_integration_tests.yml]: Sorry, my mistake, this file should not have been added to the commit as it is what I sometimes use for testing; will revert to the previous version.

**Review by philipr-za — DISMISSED** [review]

File: integration_tests/features/Reorgs.feature [hunk 2]
```diff
@@ -91,3 +91,147 @@ Feature: Reorgs
...
+    And I connect node NODE_A1 to node NODE_A3 and wait 1 seconds
+    And I connect node NODE_A2 to node NODE_A4 and wait 1 seconds
+    And I connect node SEED_A1 to node SEED_A2 and wait <SYNC_TIME> seconds
+    Then node SEED_A1 is at the same height as node SEED_A2
+    When I mine 10 blocks on SEED_A1
+    Then node SEED_A2 is at the same height as node SEED_A1
+    Then node NODE_A1 is at the same height as node SEED_A1
+    Then node NODE_A2 is at the same height as node SEED_A1
+    Then node NODE_A3 is at the same height as node SEED_A1
+    Then node NODE_A4 is at the same height as node SEED_A1
```
Comment (SWvheerden) [inline integration_tests/features/Reorgs.feature:130]: Would it not be better to replace this with the function: then all nodes are at the same height as X

**Review by SWvheerden — DISMISSED** [review]
LGTM

File: integration_tests/features/Reorgs.feature (hunk 2 above)
Comment (hansieodendaal) [reply integration_tests/features/Reorgs.feature:130]: That is an approach I considered, but it (the _all nodes_ part) would not work for chain 2 though if someone changes the number of blocks that are mined, so to keep with the pattern, I added the explicit steps.

**Pull request merged**